workflow/util/util_test.go

```diff
@@ -1026,51 +1025,6 @@ func TestRetryExitHandler(t *testing.T) {
 func TestFormulateRetryWorkflow(t *testing.T) {
 	ctx := context.Background()
 	wfClient := argofake.NewSimpleClientset().ArgoprojV1alpha1().Workflows("my-ns")
-	createdTime := metav1.Time{Time: time.Now().Add(-1 * time.Second).UTC()}
-	finishedTime := metav1.Time{Time: createdTime.Add(time.Second * 2)}
-	t.Run("Steps", func(t *testing.T) {
```

Comment:
I will check those now, the logic differed quite a lot. This sets nodes to Running I believe, whereas I delete them. 
This is needed to actually get a pod allocated, this why sometimes when  you hit retry, a workflow can be stuck on Running.

Reply:
It's not really testable easily, we cannot establish a dag for these status fields.  The root node is not present. 
This will error out complaining that the root node couldn't be found.